src/Client/Grpc/GrpcDurableEntityClient.cs

```diff
@@ -55,7 +56,18 @@ public override async Task SignalEntityAsync(
             Name = operationName,
             Input = this.dataConverter.Serialize(input),
             ScheduledTime = scheduledTime?.ToTimestamp(),
-        };
+        };
+
+        if (Activity.Current?.Id != null)
```

Comment:
I like to use pattern matching:

```suggestion
        if (Activity.Current is { } activity)
```

Then you use `activity` throughout the if block. It is guaranteed non-null there.

Reply:
But this wouldn't catch if the `Id` is null, right? I don't see how `Activity.Current` could be non-null with a null `Id`, but I figured it's better to be safe rather than sorry and keep this sort of check instead

Reply:
`Activity.Current` won't have a null Id. `Activity.Current` must always be a 'running' activity, which means non-null ID.

https://github.com/dotnet/runtime/blob/4d14dc5b4c685eeca9de610298006f4552e43955/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs#L1412